src/main/java/org/icatproject/ids/LockManager.java

```diff
@@ -134,22 +134,21 @@ public Lock lock(DsInfo ds, LockType type) throws AlreadyLockedException, IOExce
 		synchronized (lockMap) {
 			LockEntry le = lockMap.get(id);
 			if (le == null) {
-				le = new LockEntry(id, type);
+				AutoCloseable storageLock;
+				try {
+					storageLock = mainStorage.lock(ds, type == LockType.SHARED);
+				} catch (AlreadyLockedException | IOException e) {
+					throw e;
```

Comment:
The caught exception is immediatelly rethrown here.
Same behaviour as no `try-catch` block.
Add some more logic to the `catch` block (e.g. a logging) or remove the try catch…

Reply:
Indeed, you're right. This code fragment has been moved from [here](https://github.com/icatproject/ids.server/blob/703be1dcf635e43789971eb235b93ecab79143e6/src/main/java/org/icatproject/ids/LockManager.java#L151), where there was a `le.dec();` inbetween, which is not needed in the new version, because the `le` has not been incremented here in the first place.

Still hesitating to remove that though, because it documents in the code that raising `AlreadyLockedException` may also happen in the storage plugin.

Reply:
I agree, workaround could be just to add a `logger.debug` information what happened here (or actually failed).

Reply:
Ok, I finally opted for the easy solution to simply drop that uneeded try statement. In the end, it's kind of obvious that requesting a lock may fail because the resource is already locked, otherwise you wouldn't need to request the lock in the first place.